### PR TITLE
feat: benchmark viz app

### PR DESCRIPTION
**Pull request by Sayan751**
# Pull Request

### CODE REVIEW

**Comment by AureliaEffect** [comment]
The benchmarking comparison can be found here: $comparison_url

**Comment by AureliaEffect** [comment]
The benchmarking comparison can be found here: https://au2-bench-viewer.azurewebsites.net/compare?branch=master&commit=&branch=topic/benchmark-viz-app&commit=dde0f127ff2c3f630d05c9cec2a253c1657df7f5

**Comment by AureliaEffect** [comment]
The benchmarking comparison can be found here: https://au2-bench-viewer.azurewebsites.net/compare?branch=master&commit=fe528ff762832274cddd9d1203e5699551518c71&branch=topic/benchmark-viz-app&commit=a66e9935c2a352efba22eac9938e3d7caf1a2dfe

**Comment by Sayan751** [comment]
The benchmarking comparison links will not work currently, as the new client app is not yet deployed.

**Comment by AureliaEffect** [comment]
The benchmarking comparison can be found here: https://au2-bench-viewer.azurewebsites.net/compare?branch=master&commit=fe528ff762832274cddd9d1203e5699551518c71&branch=topic/benchmark-viz-app&commit=6c0b1e3aa17cacbaa56b7ffe9779edc6d7bbeaa5

**Review by bigopon — COMMENTED** [review]
It seems the address is not public? It's like this for me
image

File: test/benchmarking-apps/viewer/client/src/pages/latest-measurements/avg-measurements.ts [hunk 1]
```diff
@@ -186,7 +68,8 @@ class AvgMeasurement implements Omit<Measurement, 'browser' | 'browserVersion' |
   public durationDeleteAll: number = 0;
 
   private count = 0;
-  private _totalDuration: number;
+  @totalDuration
```
[the remark this answers is not in the document]
Comment (Sayan751) [reply test/benchmarking-apps/viewer/client/src/pages/latest-measurements/avg-measurements.ts]: I am also using it here: https://github.com/aurelia/aurelia/blob/topic/benchmark-viz-app/test/benchmarking-apps/viewer/client/src/shared/data.ts.

I could have used inheritance to share the code, but that does not really fit semantically in this case, and would have brought needless complexity. The other alternative would have been to have a simple function and encapsulate the computation in that.  IMO however, code sharing like this is not a bad idea.

Also regarding the website, it is public. Try https://au2-bench-viewer.azurewebsites.net/. Seems like a pushstate problem @fkleuver?

[the remark this answers is not in the document]
Comment (fkleuver) [reply test/benchmarking-apps/viewer/client/src/pages/latest-measurements/avg-measurements.ts]: > Also regarding the website, it is public. Try https://au2-bench-viewer.azurewebsites.net/. Seems like a pushstate problem @fkleuver?

I don't know if this is an azure or a server problem. Does it work locally when running it with the server?

[the remark this answers is not in the document]
Comment (Sayan751) [reply test/benchmarking-apps/viewer/client/src/pages/latest-measurements/avg-measurements.ts]: Yes. Runs both with webpack-dev-server as well as when hosted via the viewer/server.

**Comment by AureliaEffect** [comment]
The benchmarking comparison can be found here: https://au2-bench-viewer.azurewebsites.net/compare?branch=master&commit=2358948fbf2b82b3b37f993fa10bdd69b82769d9&branch=topic/benchmark-viz-app&commit=e876dc3574390d0141981592fbc11ec227a37621

**Pull request merged**